### PR TITLE
Feature | `graph init` Add support to download spkg files from spkg.io

### PR DESCRIPTION
**Pull request by joshuanazareth97**
## Change Description

- Allow passing a file url for spkg files instead of just a file path.
- Download the file if a url has been passed

## Related Issue / Discussion
https://github.com/graphprotocol/graph-tooling/issues/1641

## Screenshot of change
image
image
image

### CODE REVIEW

**Comment by joshuanazareth97** [comment]
Hi @saihaj can you please review this PR? Let me know if there's anything you need me to do before you merge it. Thanks!

**Comment by saihaj** [comment]
thanks @joshuanazareth97 sorry for the delay. I was away on vacation last week, catching up on things will try to land this early next week

File: .changeset/early-walls-talk.md
Comment (saihaj) [inline .changeset/early-walls-talk.md:5]:
```suggestion
---
'@graphprotocol/graph-cli': minor
---

`graph init` to allow downloading `spkg.io` files 
```

File: packages/cli/src/command-helpers/download.ts
Comment (saihaj) [inline packages/cli/src/command-helpers/download.ts:13]: I would use `fetch`, we maintain an internal wrapper that also passes some headers that can let origin servers know where the traffic is coming from.

https://github.com/graphprotocol/graph-tooling/blob/1f8d86f10ceee6f8e34ba942cfa76a37f1f44a42/packages/cli/src/fetch.ts#L5

File: packages/cli/src/command-helpers/spkg.ts
Comment (saihaj) [inline packages/cli/src/command-helpers/spkg.ts:7]: don't think we need `/streamingfast` on the subpath (for example we have semiotic on this path https://substreams.dev/semiotic-ai/uniswap-x/v0.1.0#modules)  we just need to make sure it is from `spkg.io`

we want to allow users either passing `https://spkg.io/semiotic-ai/uniswap-x-v0.1.0.spkg` or `spkg.io/semiotic-ai/uniswap-x-v0.1.0.spkg` and it should work



```suggestion
  return value.startsWith('spkg.io');
```

**Review by saihaj — CHANGES_REQUESTED** [review]
few minor nits, thanks for adding tests!

(sorry for the delayed review)

File: packages/cli/src/command-helpers/spkg.ts
Comment (joshuanazareth97) [reply packages/cli/src/command-helpers/spkg.ts:7]: Updating the check to 
```
value.startsWith(`https://spkg.io`) || value.startsWith(`spkg.io`)
```

**Comment by saihaj** [comment]
@joshuanazareth97 are you still working on this?

File: packages/cli/src/command-helpers/download.ts
Comment (joshuanazareth97) [reply packages/cli/src/command-helpers/download.ts:13]: Have now resolved this to use the wrapped fetch

**Comment by joshuanazareth97** [comment]
> @joshuanazareth97 are you still working on this?

@saihaj apologies, was occupied with some other stuff last week. Have addressed all your comments.

**Comment by joshuanazareth97** [comment]
@saihaj is there anything else you need me to do here? Can we merge this PR if the changes are made to your satisfaction?

**Comment by YaroShkvorets** [comment]
Implemented in #1788

**Pull request closed without merging**
